### PR TITLE
file sorting and download

### PR DESCRIPTION
**Pull request by davidtran**

### CODE REVIEW

File: src/components/Common/TableHeader.jsx [hunk 1]
```diff
@@ -30,11 +30,11 @@ function TableHeader({ isSelected, onSelectAll }) {
...
           </Button>
         </th>
-        <th scope="col" width="50%">
+        <th className="sortable" scope="col" width="50%" onClick={() => handleSortFiles('name')}>
           File Name
         </th>
-        <th scope="col">Date Added</th>
-        <th scope="col">Size</th>
+        <th className="sortable" scope="col" onClick={() => handleSortFiles('createdDate')}>Date Added</th>
+        <th className="sortable" scope="col" onClick={() => handleSortFiles('size')}>Size</th>
```
Comment (qubeeio) [inline src/components/Common/TableHeader.jsx:37]: Can we add arrow indicators using FontAwesome that'll tell us if we're sorting in ASC or DESC in each column?

File: backend/routes/files.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const mimeTypes = require("../utils/mimetypes");
 const router = express.Router();
 const { File, fileBasePath, validate } = require("../models/file");
 
-const tempDir = join(__dirname, "../public", "uploads/");
+const tempDir = join(__dirname, "../../public", "uploads/");
```
Comment (qubeeio) [inline backend/routes/files.js:16]: This `tempDir` should not point to the root public directory. Instead, create an uploads folder in the backend directory: `backend > uploads`. I forgot to include that folder during the initial seeding. I will be fixing that on my next update.

File: backend/routes/files.js [hunk 3]
```diff
@@ -187,4 +187,27 @@ router.get("/:id", async (req, res) => {
...
+    }));
+
+  try {
+    await res.zip({
+      files: zipFiles,
+      filename: 'download.zip'
+    });
+  } catch (e) {
+    console.log(e);
+  }
```
Comment (qubeeio) [inline backend/routes/files.js:209]: Is there a way we can download the actual file if it's a single file selected for download (i.e. img_123.jpg), but a zip file if it's 2 or more files selected for download?

Also, let's rename the zip file from **download.zip** to **QubeeFiles.zip**

File: backend/routes/files.js (hunk 2 above)
Comment (davidtran) [reply backend/routes/files.js:16]: ok cool, I got an error when start the app, so I thought the path is not correct.

File: backend/routes/download.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    try {
+      await res.zip({
+        files: zipFiles,
+        filename: 'QubeeFiles.zip'
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  } else if (availableFiles.length === 1 ) {
+    const filename = join(uploadsFolder, file.slug);
```
Comment (qubeeio) [inline backend/routes/download.js:31]: I'm getting this error here every time I tried downloading a single file. I also get that error when I try downloading multiple files, but the files are downloaded regardless after getting that error on multiple files.
Screen Shot 2020-07-06 at 5 49 46 PM

File: src/components/Common/TableHeader.jsx [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
 
+  function renderSortDirection(attributeName) {
+    if (!sorting || sorting.attribute !== attributeName) return null;
+    if (sorting.direction === 'ASC') {
+      return <FontAwesomeIcon icon="arrow-up" />;
+    } else {
+      return <FontAwesomeIcon icon="arrow-down" />;
+    }
+  }
+
```
Comment (qubeeio) [inline src/components/Common/TableHeader.jsx:21]: Can we make the default be ASC (A to Z) order with the arrow pointing down?

File: backend/routes/download.js (hunk 4 above)
Comment (davidtran) [reply backend/routes/download.js]: @qubeeio I have fixed this error

File: src/components/Common/TableHeader.jsx (hunk 5 above)
Comment (davidtran) [reply src/components/Common/TableHeader.jsx]: done

**Comment by Undercover-Geek** [comment]
Well done @davidtran, I've merged your updates to master. We have more updates coming down the pipeline. Joshua will be in touch with you soon.

**Pull request merged**